### PR TITLE
Feature/#57 switch build system from maven to gradle

### PR DESCRIPTION
**Pull request by DivineThreepwood**
### :scroll: Description

Changes proposed in this pull request:
* switch build system from maven to gradle
* setup ci to
  * inspect code
  * build and test
  * publish to maven repos
* cleanup maven stuff

### CODE REVIEW

**Comment by DivineThreepwood** [comment]
Should not be merged yet since only a single artifact is deployed instead of each submodule. Still some investigation needed. However, the code review can already be performed.

**Comment by DivineThreepwood** [comment]
All open issues are solved by not overwriting the artifact id for each submodule.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
```
Comment (pLeminoq) [inline gradle/wrapper/gradle-wrapper.properties:2]: Is it common to have to gradle wrappers in the repository? A second one was added in the buildSrc directorey.

File: module/processing/json/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+dependencies {
+    api(project(":jul.processing"))
+    api("com.fasterxml.jackson.core:jackson-core:[2.11,2.12-alpha)")
+    api("com.fasterxml.jackson.core:jackson-databind:[2.11,2.12-alpha)")
+    api(project(":jul.pattern"))
+    api("com.googlecode.protobuf-java-format:protobuf-java-format:_")
+}
+
+description = "JUL Processing JSon"
```
Comment (pLeminoq) [inline module/processing/json/build.gradle.kts]: Not JSON instead of JSon?

**Review by pLeminoq — COMMENTED** [review]
Looks good. Remove the second gradle wrapper and it will be approved.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (DivineThreepwood) [reply gradle/wrapper/gradle-wrapper.properties:2]: has been removed

File: module/processing/json/build.gradle.kts (hunk 2 above)
Comment (DivineThreepwood) [reply module/processing/json/build.gradle.kts]: Requested changes applied.

**Review by pLeminoq — APPROVED** [review]

**Pull request merged**